azure/src/main/java/org/apache/iceberg/azure/adlsv2/ADLSLocation.java

```diff
@@ -53,19 +63,17 @@ class ADLSLocation {
 
     ValidationException.check(matcher.matches(), "Invalid ADLS URI: %s", location);
 
-    String authority = matcher.group(1);
-    String[] parts = authority.split("@", -1);
-    if (parts.length > 1) {
-      this.container = parts[0];
-      this.storageAccount = parts[1];
-    } else {
-      this.container = null;
-      this.storageAccount = authority;
+    try {
+      URI uri = new URI(location);
+      this.container = uri.getUserInfo();
+      // storage account name is the first part of the host
+      int accountSplit = uri.getHost().indexOf('.');
+      String storageAccountName = uri.getHost().substring(0, accountSplit);
+      this.storageAccount = String.format("%s.dfs.core.windows.net", storageAccountName);
+      this.path = uri.getPath().length() > 1 ? uri.getRawPath().substring(1) : "";
+    } catch (URISyntaxException e) {
+      throw new ValidationException("Invalid URI: %s", location);
```

Comment:
This is validation we didn't have before correct?

Reply:
I went with `IllegalArgumentException` b/c that seems more appropriate for this.  Now it throws the same exception if the RegEx doesn't match or if the URI is malformed.